matlab/test/arrow/type/tDateTypeTest.m

```diff
@@ -0,0 +1,80 @@
+% Shared superclass for DateType-related tests for Date32Type and Date64Type
+classdef tDateType < matlab.unittest.TestCase
+    
+    properties
```

Comment:
Additionally, if this test class inherits from `hFixedWidthType`, you won't need to declare `ArrowType`, `TypeID`, `BitWidth`, or `ClassName`. These properties are declared as `Abstract` in `hFixedFixedType` and are concretely defined in `tDate32Type` and `tDate64Type`.

I would also recommend declaring the `ConstructionFcn` `Abstract` so that child classes of this one must declare that property.